### PR TITLE
Selected input scenarios for basic joins and advance scenarios for joiner

### PR DESCRIPTION
**Pull request by rahuldash171**
Tests Covered :
Advance Inner Join
Advance Outer Join
Basic Inner Join with selected inputs 
Basic outer join with selected inputs and required input

### CODE REVIEW

File: core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/locators/JoinerLocators.java [hunk 1]
```diff
@@ -28,7 +28,17 @@ public class JoinerLocators {
 
   public static WebElement fieldAliasCheckBox(String pluginName, String field) {
     String xpath = "//*[@data-cy='" + pluginName + "-stage-expansion-panel']" + "//*[@data-cy='" + field +
-      "-field-selector-name']/..//*[@type='checkbox']";
+      "-field-selector-name']/..//*[@data-cy='"+ field +"-field-selector-checkbox']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement requiredInputCheckboxFirstFile(String pluginName){
```
Comment (poojantcs) [inline core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/locators/JoinerLocators.java]: Update: pluginName to input. 
Remove the second locator and use the index as a parameter.

File: core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/stepsdesign/Joiner.java [hunk 2]
```diff
@@ -36,6 +36,16 @@ public void uncheckPluginFieldAliasCheckBox(String plugin, String field) {
     JoinerActions.uncheckPluginFieldAliasCheckBox(plugin, field);
   }
 
+  @Then("Click on the required input checkbox for first schema {string}")
```
Comment (poojantcs) [inline core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/stepsdesign/Joiner.java]: Add index and remove the second steps

File: core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/locators/JoinerLocators.java (hunk 1 above)
Comment (rahuldash171) [reply core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/locators/JoinerLocators.java]: Done.

File: core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/stepsdesign/Joiner.java (hunk 2 above)
Comment (rahuldash171) [reply core-plugins/src/e2e-test/java/io/cdap/plugin/joiner/stepsdesign/Joiner.java]: Done , removed second and used index

**Pull request closed without merging**